textdb/textdb-api/src/main/java/edu/uci/ics/textdb/api/storage/ICatalogManager.java

```diff
@@ -0,0 +1,36 @@
+package edu.uci.ics.textdb.api.storage;
+
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.common.Schema;
+
+/**
+ * ICatalogManager is the interface for TextDB's catalog manager, 
+ *   which handles the meta data information about each table.
+ * 
+ * @author Zuozhi Wang
+ */
+public interface ICatalogManager {
+    
+    public ICatalogManager getCatalogManager();
+    
+    public boolean tableExists(String tableName);
+    
+    // create a new table, tableName must be unique
+    public void createTable(String tableName, String indexDirectory, Schema schema) throws Exception;
+    
+    // drop a table
+    public void deleteTable(String tableName) throws Exception;
+    
+    // insert a tuple to a table, the primaryAttribute field must be unique
+    public void insertTuple(String tableName, ITuple tuple, String primaryAttribute) throws Exception;
+    
+    // delete a tuple by its primary attribute
+    public void deleteTuple(String tableName, String primaryAttribute, Object value) throws Exception;
```

Comment:
Per our discussion, we decided the following:

- The user does NOT need to provide a primary key attribute;
- The system always has a primary key attribute, say, "_id", as in Mongo, which is automatically generated or incremented.
- Lucene supports efficient search on this key.
- The function `addTuple()` should return this generated key to the caller.
- To keep the generality of the type for this primary key, let's introduce a class or field type for the type of the primary key, and also its default name such as "_id".